sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1126 @@
+            Iterator<DetectLanguageResult> responseItem = response.getValue().iterator();
+            if (responseItem.hasNext()) {
+                return Mono.just(new SimpleResponse<>(response, responseItem.next()));
+            }
+            return monoError(logger, new RuntimeException("Unable to retrieve language for the provided text."));
+        });
+    }
+
+    /**
+     * Returns the detected language for a batch of input.
+     *
+     * @param inputs The list of texts to be analyzed.
+     * @return A {@link Mono} containing the {@link DocumentResultCollection batch} of the
+     * {@link DetectLanguageResult detected languages}.
+     * @throws NullPointerException if {@code inputs} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<DocumentResultCollection<DetectLanguageResult>> detectLanguages(List<String> inputs) {
+        return detectLanguagesWithResponse(inputs, null).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Returns the detected language for a batch of input with the provided country hint.
+     *
+     * @param inputs The list of texts to be analyzed.
+     * @param countryHint A country hint for the entire batch. Accepts two letter country codes specified by ISO 3166-1
+     * alpha-2. Defaults to "US" if not specified.
+     * @return A {@link Response} of {@link Mono} containing the {@link DocumentResultCollection batch} of the
+     * {@link DetectLanguageResult detected languages}.
+     * @throws NullPointerException if {@code inputs} is {@code null}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DocumentResultCollection<DetectLanguageResult>>> detectLanguagesWithResponse(
+        List<String> inputs, String countryHint) {
+        try {
+            return withContext(context -> detectLanguagesWithResponse(inputs, countryHint, context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<DocumentResultCollection<DetectLanguageResult>>> detectLanguagesWithResponse(List<String> inputs,
+                                                                                               String countryHint,
+                                                                                               Context context) {
+        List<DetectLanguageInput> languageInputs = getLanguageInputList(inputs, countryHint);
+        return detectBatchLanguagesWithResponse(languageInputs, null, context);
+    }
+
+    /**
+     * Helper method to convert text list input to LanguageInput.
+     *
+     * @param inputs the list of user provided texts.
+     * @param countryHint the countryHint provided by user for texts.
+     *
+     * @return the LanguageInput list objects to provide the service.
+     */
+    private static List<DetectLanguageInput> getLanguageInputList(List<String> inputs, String countryHint) {
```

Comment:
Generally, you organise methods by visibility to make it easier for people to reason about. public methods, package-private then private.